src/test/java/com/iexec/worker/tee/post/compute/web2/UploaderServiceTests.java

```diff
@@ -0,0 +1,162 @@
+package com.iexec.worker.tee.post.compute.web2;
+
+import com.dropbox.core.DbxException;
+import com.dropbox.core.DbxRequestConfig;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.users.DbxUserUsersRequests;
+import com.dropbox.core.v2.users.FullAccount;
+import com.iexec.worker.tee.post.compute.PostComputeException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import static com.iexec.common.replicate.ReplicateStatusCause.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+class UploaderServiceTests {
+
+    private static final String TASK_ID = "0x0";
+    private static final String DROPBOX_TOKEN = "DROPBOX_TOKEN";
+    private static final String IPFS_TOKEN = "IPFS_TOKEN";
+    private static final String REMOTE_FILENAME = "remoteFileName.zip";
+    private static final String REMOTE_FILEPATH = "https//foo.bar/remoteFileName.zip";
+
+    @TempDir
+    File tmpFolder;
+
+    @Mock
+    DropBoxService dropBoxService;
+
+    @Spy
+    @InjectMocks
+    UploaderService uploaderService;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    //region createDropboxClient
+    @Test
+    void shouldCreateDropboxClient() {
+        assertNotNull(uploaderService.createDropboxClient(DROPBOX_TOKEN, DbxRequestConfig.newBuilder("").build()));
+    }
+    //endregion
+
+    //region uploadToDropBox
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(strings = {"", "accountId"})
+    void shouldUploadToDropbox(String accountId) throws DbxException, IOException {
```

Comment:
Maybe we can keep `doReturn` only where it is required:
```
    void shouldUploadToDropbox(String accountId) throws DbxException, IOException {
        final String fileToUpload = Files.createFile(Path.of(tmpFolder.getAbsolutePath(), "fileToUpload.zip")).toString();

        final DbxClientV2 client = mock(DbxClientV2.class);
        final DbxUserUsersRequests users = mock(DbxUserUsersRequests.class);
        final FullAccount account = mock(FullAccount.class);

        when(client.users()).thenReturn(users);
        when(users.getCurrentAccount()).thenReturn(account);
        when(account.getAccountId()).thenReturn(accountId);

        doReturn(client).when(uploaderService).createDropboxClient(eq(DROPBOX_TOKEN), any());
        when(dropBoxService.uploadFile(eq(client), any(), eq("/results/" + REMOTE_FILENAME))).thenReturn(REMOTE_FILEPATH);

        final String actualFilePath = assertDoesNotThrow(() -> uploaderService.uploadToDropBox(fileToUpload, DROPBOX_TOKEN, REMOTE_FILENAME));
        assertEquals(REMOTE_FILEPATH, actualFilePath);
    }
```

Reply:
Sure, see https://github.com/iExecBlockchainComputing/tee-worker-post-compute/pull/50/commits/58d44b5e85a6cd662e4c09ee2c4b2eae8aebaba1.

Reply:
Thank you :pray: